File: app/src/main/java/com/pabji/androidappmovie/data/net/requestsModels/SearchRequest.java
package com.pabji.androidappmovie.data.net.requestsModels;

import java.util.HashMap;
import java.util.Map;


public class SearchRequest extends BaseRequest{

    public static final String LANGUAGE = "en";
    public static final Boolean IS_NAME_REQUIRED = true;
    public static final String STYLE = "FULL";

    private String cityName;
    private Integer maxRows = 20;
    private Integer startRow = 0;

    public SearchRequest(String cityName) {
        this.cityName = cityName;
    }

    public SearchRequest(String cityName, Integer maxRows, Integer startRow) {
        this.cityName = cityName;
        this.maxRows = maxRows;
        this.startRow = startRow;
    }

    @Override
    public Map<String, String> toQueryMap() {
        Map<String, String> data = new HashMap<>();
        data.put("q", cityName);
        data.put("maxRows", String.valueOf(maxRows));
        data.put("startRow", String.valueOf(startRow));
        data.put("lang", LANGUAGE);
        data.put("isNameRequired:", String.valueOf(IS_NAME_REQUIRED));
        data.put("style", STYLE);
        data.put("username", getUsername());

        return data;
    }
}

File: app/src/main/java/com/pabji/androidappmovie/data/net/requestsModels/SearchRequest.kt
package com.pabji.androidappmovie.data.net.requestsModels


data class SearchRequest(var api_key : String, var language : String? = "es", var page : Int? = 1){


    fun toQueryMap() : Map<String,String> {
        val data = hashMapOf<String, String>()
        data["api_key"] = api_key

        language?.let {
            data["language"] = it
        }

        page?.let {
            data["page"] = it.toString()
        }

        return data
    }
}
